sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -138,22 +139,35 @@ export class InstitutionController extends BaseController {
       );
     }
 
-    // Get user details
-    const accountDetails = await this.accountService.getAccountDetails(
-      body.userId,
-    );
-    if (!accountDetails) {
-      throw new UnprocessableEntityException(
-        `Unable to account detail of user ${body.userId}`,
-      );
+    let userEntity: User;
+
+    if (body.userGuid) {
+      userEntity = await this.userService.getUser(`${body.userGuid}@bceid`);
     }
 
-    // Create User
-    const userEntity = this.userService.create();
-    userEntity.email = accountDetails.user.email;
-    userEntity.firstName = accountDetails.user.firstname;
-    userEntity.lastName = accountDetails.user.surname;
-    userEntity.userName = `${accountDetails.user.guid}@bceid`;
+    if (!userEntity) {
+      // Get user details
+      const accountDetails = await this.accountService.getAccountDetails(
```

Comment:
This method is supposed to be called all the time, right?
Why is it being called only if "userEntity" is not defined?

Reply:
Please check code comments.

Reply:
Still not clear to me. We should always check for "Get user details from BCeID". Why is it inside an "if (!userEntity)"?

Reply:
BCeID service call is require to get user info... if same user already exits we skip that call and create auth association.

Reply:
How do we ensure that the user belongs to the institution that it is being associated with?